src/webview_provider/docsEditPanel.ts

```diff
@@ -266,12 +266,77 @@ export class DocsEditViewPanel implements WebviewViewProvider {
     return (metadata as TestMetadataAcceptedValues).values !== undefined;
   }
 
+  private getTestDataByModel(message: any, modelName: string) {
+    const tests = message.updatedTests as undefined | TestMetaData[];
+
+    if (!tests?.length) {
+      this.terminal.debug(
+        "docsEditViewPanel:getTestDataByModel",
+        "No test data passed",
+      );
+      return;
+    }
+
+    const updatedTests = tests.filter((test) => {
+      const modelNameInTest = test.test_metadata?.kwargs.model;
+      if (test.column_name || !modelNameInTest) {
+        return false;
+      }
+      if (modelNameInTest === modelName) {
+        return true;
+      }
+      // model name could be {{ get_where_subquery(ref('dim_hosts_cleansed')) }}
+      if (modelNameInTest.match(/'([^']+)'/)?.[1] === modelName) {
```

Comment:
didn't we find a more reliable way of doing this?

Reply:
unfortunately no. dbt wraps it by default and i beleive atleast the model name will be encapsulated within quotes, so implemented in this way